tools/src/main/scala/scala/scalanative/codegen/Lower.scala

```diff
@@ -522,8 +525,19 @@ object Lower {
             let(n, Op.Copy(Val.Global(impl, Type.Ptr)), unwind)
           case _ =>
             obj.ty match {
-              case ClassRef(cls) =>
+              // It may happen that method exists in multiple targets, but is not a virtual call
+              // In such cases we handle it as statically resolvable method with single target
+              case ClassRef(cls) if cls.calls.contains(sig) =>
```

Comment:
I think the proposed fix is correct when everything else is correct. However, I fear that it will too easily "tolerate" other errors without complaining, and instead generate wrong code.

We should be able to determine in advance if a method *should* be a virtual call or not. And if it should not be a virtual call (because it's a constructor, for example), then don't even collect the `targets`. For a non-virtual call, we should be able to find with certainty a *single* `target`, and generate a call to that. Only if we have a virtual call should we collect the multiple `targets`, then generate the proper code for `Seq()`, `Seq(impl)` or more.

Reply:
I've changed logic to explicitly check if method is virtual or not with additional handling Null (short-circuit throwing null)